iroh-gateway/src/handlers.rs

```diff
@@ -180,12 +180,18 @@ async fn request_preprocessing<T: ContentLoader + Unpin>(
         }
     }
 
-    let path_metadata = state
-        .client
-        .retrieve_path_metadata(path.clone())
-        .await
-        .map_err(|e| GatewayError::new(StatusCode::INTERNAL_SERVER_ERROR, &e))?;
-    // TODO: handle 404 or error
+    let path_metadata = match state.client.retrieve_path_metadata(path.clone()).await {
```

Comment:
@b5 this is the crucial bit - the earliest moment we ping store/p2p. I've updated error responses from `fetch_gateway` and `fetch_bitswap` so that it now catches them and returns a more sane response (`failed to find {cid}` || `offline`). 
We should do similar for `fetch_store` to make it less jarring to the end users.